Removing some cruff and updating signatures for consistency

drisspg opened the pull request:
# Summary

- This removes a bunch of example score mods that were primarily used for testing and places them directly in the test file. We should follow up with merging test_flex_decode and test_flash when the velocity slows down a little
- Fixes a bug with indexing on block mask
- Adds some doc strings to helper funcs and fixes some misc typing things
- Forces functions passed to `create_block_mask` to mask_mods and updates tests files

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #131013
* #131012
* #130904
* __->__ #130871



cc @albanD @mruberry @jbschlosser @walterddr @mikaylagawarecki @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang

drisspg committed:
Removing some cruff and updating signatures for consistency

[ghstack-poisoned]

drisspg committed:
Update on "Removing some cruff and updating signatures for consistency"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

drisspg committed:
Update on "Removing some cruff and updating signatures for consistency"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

drisspg commented on torch/nn/attention/flex_attention.py:
```diff
@@ -302,20 +301,34 @@ def __getitem__(self, index) -> "BlockMask":
             mask_mod=self.mask_mod,
         )
 
+    def __repr__(self):
```
I found this helpful for debugging but its a big repr

drisspg committed:
Update on "Removing some cruff and updating signatures for consistency"




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

drisspg commented on torch/nn/attention/flex_attention.py:
```diff
@@ -295,27 +294,51 @@ def __str__(self):
     def __getitem__(self, index) -> "BlockMask":
         new_kv_num_blocks = self.kv_num_blocks[index]
         new_kv_indices = self.kv_indices[index]
+        new_kv_num_blocks_full = (
```
cc @Chillee  I think this a bug from lower in the stack

Chillee replied:
mm.... yes, this is a bug. Do you want me to fix it lower down in the stack?

drisspg replied:
whatever makes it easiest to land lol

Chillee commented on torch/nn/attention/flex_attention.py:
```diff
@@ -544,12 +561,22 @@ def create_mask(
-# presence of the torchfunctionmdoe
 def _create_block_mask_inner(
-    mod_fn, B, H, M, N, device, KV_BLOCK_SIZE, Q_BLOCK_SIZE, mod_type
+    mod_fn,
```
Can rename this to `mask_mod`

drisspg replied:
will do

drisspg committed:
Update on "Removing some cruff and updating signatures for consistency"

# Summary

- This removes a bunch of example score mods that were primarily used for testing and places them directly in the test file. We should follow up with merging test_flex_decode and test_flash when the velocity slows down a little
- Fixes a bug with indexing on block mask
- Adds some doc strings to helper funcs and fixes some misc typing things
- Forces functions passed to `create_block_mask` to mask_mods and updates tests files




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

drisspg commented:
@pytorchbot merge

drisspg committed:
Update on "Removing some cruff and updating signatures for consistency"

# Summary

- This removes a bunch of example score mods that were primarily used for testing and places them directly in the test file. We should follow up with merging test_flex_decode and test_flash when the velocity slows down a little
- Fixes a bug with indexing on block mask
- Adds some doc strings to helper funcs and fixes some misc typing things
- Forces functions passed to `create_block_mask` to mask_mods and updates tests files




cc albanD mruberry jbschlosser walterddr mikaylagawarecki voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx peterbell10 ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang

[ghstack-poisoned]

drisspg commented:
@pytorchbot merge -I

drisspg commented:
@pytorchbot merge -i

drisspg commented:
@pytorchbot merge -f "timeout"
